pkg/checkoutinfo/checkoutinfo.go

```diff
@@ -0,0 +1,229 @@
+package checkoutinfo
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/go-openapi/strfmt"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/buildscript"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model/buildplanner"
+)
+
+type configurer interface {
+	GetBool(string) bool
+}
+
+type projecter interface {
+	Owner() string
+	Name() string
+	BranchName() string
+	LegacyCommitID() string
+	SetLegacyCommit(string) error
+	Dir() string
+	URL() string
+}
+
+type CheckoutInfo struct {
+	auth    *authentication.Auth
+	cfg     configurer
+	project projecter
+}
+
+var ErrBuildscriptNotExist = errors.New("Build script does not exist")
+
+type ErrInvalidCommitID struct {
+	CommitID string
+}
+
+func (e ErrInvalidCommitID) Error() string {
+	return "invalid commit ID"
+}
+
+func New(auth *authentication.Auth, cfg configurer, project projecter) *CheckoutInfo {
+	return &CheckoutInfo{auth, cfg, project}
+}
+
+func (c *CheckoutInfo) Owner() string {
+	return c.project.Owner()
+}
+
+func (c *CheckoutInfo) Name() string {
+	return c.project.Name()
+}
+
+func (c *CheckoutInfo) Branch() string {
+	return c.project.BranchName()
+}
+
+func (c *CheckoutInfo) CommitID() (strfmt.UUID, error) {
+	if c.cfg.GetBool(constants.OptinBuildscriptsConfig) {
+		if script, err := c.BuildScript(); err == nil {
+			commitID, err2 := script.CommitID()
+			if err2 != nil {
+				return "", errs.Wrap(err, "Could not get commit ID from build script")
+			}
+			return commitID, nil
+		} else if !errors.Is(err, ErrBuildscriptNotExist) {
+			return "", errs.Wrap(err, "Could not get build script")
+		}
+		// Fall back on activestate.yaml.
```

Comment:
If the buildscript exists then I think we should just fail here. Falling back to the as.yaml at that point is working around some bug.